examples/src/components/DrillWithExternalDataExample.jsx

```diff
@@ -0,0 +1,378 @@
+            employeeTableFilters.push({
+                positiveAttributeFilter: {
+                    displayForm: {
+                        identifier: locationStateDisplayFormIdentifier
+                    },
+                    in: [state.uri]
+                }
+            });
+        }
+        if (location) {
+            employeeTableFilters.push({
+                positiveAttributeFilter: {
+                    displayForm: {
+                        identifier: locationNameDisplayFormIdentifier
+                    },
+                    in: [location.uri]
+                }
+            });
+        }
+        const employeeTable = (
+            <div style={{ height: 300 }} className="s-employee-table" >
+                <Table
+                    projectId={projectId}
+                    measures={[averageDailySalesMeasure]}
+                    attributes={[employeeNameAttribute]}
+                    drillableItems={[{
+                        identifier: employeeNameIdentifier
+                    }]}
+                    filters={employeeTableFilters}
+                    onFiredDrillEvent={this.onEmployeeDrill}
+                />
+            </div>
+        );
+
+        const totalSalesMeasure = {
+            measure: {
+                localIdentifier: 'totalSales',
+                definition: {
+                    measureDefinition: {
+                        item: {
+                            identifier: totalSalesIdentifier
+                        }
+                    }
+                },
+                alias: 'Total Sales'
+            }
+        };
+
+        const salesTableFilters = [];
+        if (state) {
+            salesTableFilters.push({
+                positiveAttributeFilter: {
+                    displayForm: {
+                        identifier: locationStateDisplayFormIdentifier
+                    },
+                    in: [state.uri]
+                }
+            });
+        }
+        if (location) {
+            salesTableFilters.push({
+                positiveAttributeFilter: {
+                    displayForm: {
+                        identifier: locationNameDisplayFormIdentifier
+                    },
+                    in: [location.uri]
+                }
+            });
+        }
+
+        const totalSalesTable = (
+            <div style={{ height: 300 }} className="s-sales-chart">
+                <ColumnChart
+                    projectId={projectId}
+                    measures={[totalSalesMeasure]}
+                    viewBy={locationNameAttribute}
+                    filters={salesTableFilters}
+                    drillableItems={[{
+                        identifier: locationNameDisplayFormIdentifier
+                    }]}
+                    onFiredDrillEvent={this.onLocationDrill}
+                />
+            </div>
+        );
+
+        const { employee, employee3rdPartyData } = this.state;
+
+        let employeeDetails = null;
```

Comment:
I'd move this to separate render function which would return the component according to employee3rdPartyData:

```
renderEmployeeDetails(employee) {
  if (employee3rdPartyData.isError) {
           return <ErrorComponent height={150} message="There was an error getting employee details" />;
  } else if (employee3rdPartyData.isLoading) {
         return  <LoadingComponent height={150} />;
      } else if (employee3rdPartyData.data) {
         return <EmployeeProfile {...employee3rdPartyData.data} />;
      }

     return null;
}